lib/plug/telemetry.ex

```diff
@@ -0,0 +1,31 @@
+defmodule Plug.Telemetry do
+  @moduledoc """
+  A plug to instrument the pipeline with `:telemetry` events.
+  """
+
+  @behaviour Plug
+
+  @impl true
+  def init(_), do: []
```

Comment:
The plug should receive the `:event_prefix` as option, such as:

    plug Plug.Telemetry, event_prefix: [:phoenix, :endpoint]

It should be required. Then we can compute both start and stop events on init and pass it forward as a state in the tuple.

Reply:
Why is this option required? To differentiate between events fired by multiple "instances" of the plug?

Reply:
If the event name is configurable, how will a reporter know how to attach to it? 

From my perspective, it'd be nice if a reporter was able to attach itself to a standard set of events and be able to do something with them out of the box (not to the exclusion of other events).